### ISSUE TITLE
Prune chapter arrays [Feature request]

### ISSUE THREAD

**Issue by cpiber** [Glagan/MyMangaDex#50]
If you read a lot (like me) and have 'Save all opened chapters' enabled (like me), chances are that after some time, you'll run out of space in your localstorage, where MMD saves it's data.
Therefore I suggest to add try/catch blocks everywhere storage is updated to be able to display a notification. This notification should then hint one to prune the chapter arrays, which might be implemented as a button on the options page and simply sets all chapter arrays to one with only the most recent in it.
Another idea would be to completely clear the chapters array on all titles that are not in the reading list, since completed/on-hold/dropped titles don't need to retain everything. This could be automatically done when moving lists and/or a separate button on the options page.

**Comment by cpiber** [comment]
Btw, I would implement this if you're fine with it, maybe we'd first chat to make sure we're both on the same page, since deleting stuff is always an important change.

**Comment by Glagan** [comment]
There already is something to prune the oldest chapters in `insertChapter` [https://github.com/Glagan/MyMangaDex/blob/b5257f75ba47114beecbf07d2b6d165cf222be35/scripts/myMangaDex.js#L1066](https://github.com/Glagan/MyMangaDex/blob/b5257f75ba47114beecbf07d2b6d165cf222be35/scripts/myMangaDex.js#L1066)
And this function (I think) is already used so it should automatically do it's thing.

But I don't remember, maybe I removed the `maxChapterSaved` options since I didn't think we will run into a space issue, and it defaults to 500.

Maybe just adding back the options (if it's removed) and set it lower (500 is too much anyway) should fix the space issue.  
But there also should be a bit of code to remove chapters above the limit when the option is updated, I don't think I did that before.
